docs/source/export.rst

```diff
@@ -1,11 +1,292 @@
+  64] and [32, 128] shaped tensors that we passed in as example input. ``s0``
+  and ``s1`` are symbolic shapes which represents a range of values.
+
+* ``exported_program.range_constraints`` describes the ranges of each symbolic
+  shape appearing in the graph. In this case, we see that ``s0`` and ``s1`` have
+  the range [2, inf]. For technical reasons that are difficult to explain here,
+  they are assumed to be not 0 or 1. This is not a bug, and does not necessarily
+  mean that the exported program will not work for dimensions 0 or 1. See
+  `The 0/1 Specialization Problem <https://docs.google.com/document/d/16VPOa3d-Liikf48teAOmxLc92rgvJdfosIy-yoT38Io/edit?fbclid=IwAR3HNwmmexcitV0pbZm_x1a4ykdXZ9th_eJWK-3hBtVgKnrkmemz6Pm5jRQ#heading=h.ez923tomjvyk>`_
+  for an in-depth discussion of this topic.
+
+* ``exported_program.equality_constraints`` describes which dimensions are
+  equivalent. Since we specified in the constraints that the first dimension of
+  each argument is equivalent,
+  (``dynamic_dim(example_args[0], 0) == dynamic_dim(example_args[1], 0)``),
+  we see in the equality constraints the tuple specifying that ``arg4_1``
+  dimension 0 and ``arg5_1`` dimension 0 are equal.
+
+
+Limitations of torch.export
+---------------------------
+
+Graph Breaks
+^^^^^^^^^^^^
+
+As ``torch.export`` is a one-shot process for capturing a computation graph from
+a PyTorch program, it will ultimately run into untraceable parts of programs as
+it is nearly impossible to support tracing all PyTorch and Python features. In
+the case of ``torch.compile``, untraceable parts of programs will "graph break"
+and run the unsupported operation with default Python evaluation. However,
+``torch.export`` will require users to provide additional information or rewrite
+parts of their code to make it traceable. As the foundation of the tracing is
+based on TorchDynamo, which evaluates at the Python bytecode level, there
+will be significantly fewer rewrites required compared to previous tracing
+frameworks.
+
+When a graph break is encountered, :ref:`ExportDB <torch.export_db>` is a great
+source of supported and unsupported programs, along with ways to rewrite
+programs to make them traceable.
+
+Control Flow
+^^^^^^^^^^^^
+
+Graph breaks can also be encountered on general control flow
+(ex. ``if x[0] > 4``), which a tracing compiler cannot possibly deal with
+without generating code for a combinatorially exploding number of paths. In such
+cases, users will need to rewrite their code using special
+**Control Flow Operators**.
+
+Data-Dependent Accesses
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Data dependent behavior such as using the value inside of a tensor to construct
+another tensor, or using the value of a tensor to slice into another tensor, is
+also something the tracer cannot fully determine. Users will need to rewrite
+their code using the inline constraint APIs
+:func:`torch.export.constrain_as_size` and
```

Comment:
Eh, we really should move these out of export. Just call the underlying ops directly (easy to change because there are no current users).